database/sql/src/main/scala/cromwell/database/slick/tables/WorkflowStoreEntryComponent.scala

```diff
@@ -73,31 +73,96 @@ trait WorkflowStoreEntryComponent {
   )
 
   /**
-    * Returns up to "limit" startable workflows, sorted by submission time.
-    */
-  def fetchStartableWorkflows(limit: Long,
-     heartbeatTimestampTimedOut: Timestamp,
-     excludeWorkflowState: String,
-     excludedGroups: Set[String]
-    ): Query[WorkflowStoreEntries, WorkflowStoreEntry, Seq] = {
-      val query = for {
-        row <- workflowStoreEntries
-        /*
+   * Return hog group with the lowest count of actively running workflows
+   */
+  def getHogGroupWithLowestRunningWfs(heartbeatTimestampTimedOut: Timestamp,
+                                      excludeWorkflowState: String,
+                                      excludedGroups: Set[String]): Query[Rep[Option[String]], Option[String], Seq] = {
+    val startableWorkflows = for {
+      row <- workflowStoreEntries
+      /*
         This looks for:
-
         1) Workflows with no heartbeat (newly submitted or from a cleanly shut down Cromwell).
         2) Workflows with old heartbeats, presumably abandoned by a defunct Cromwell.
+        3) Workflows not in "OnHold" state
+        4) Workflows that don't belong to hog groups in excludedGroups
+      */
+      if (row.heartbeatTimestamp.isEmpty || row.heartbeatTimestamp < heartbeatTimestampTimedOut) &&
+        (row.workflowState =!= excludeWorkflowState) &&
+        !(row.hogGroup inSet excludedGroups)
+    } yield row
+
+    // calculates the count of startable workflows per hog group and oldest submission timestamp of workflow
+    // present in that hog group
+    val numOfStartableWfsByHogGroup = startableWorkflows
+      .groupBy(_.hogGroup)
+      .map { case (hogGroupName, groups) => (hogGroupName, groups.length, groups.map(_.submissionTime).min) }
```

Comment:
I guess I had named it groups because of `groupBy` and they were groups of workflows  based on `hogGroup`. But `workflows` make more sense now because of `submissionTime` being involved.